### PR TITLE
feat: update tx type and counterparty mapping

### LINKED ISSUE #5716: Limit counterparty options in the UI
## Abstract

Not the best UX in the window for editing the decoded events. In `counterparty` field users are able to input any text value, which may not give the results that they would intuitively expect. 

## Specification

After #5689 change the allowed options for counterparty-related fields in the UI. The current idea is to do it in the following way:

- [ ] Limited `protocol` to the list of supported protocols, and ideally warn users if the selected protocol doesn't make any change to the accounting logic.

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #5716

## Checklist

- [x] Add balancer protocol
- [x] Add some type-subtype combination in frontend.
- [x] Limit counterparty list in UI

### CODE REVIEW

File: frontend/app/src/components/history/TransactionEventForm.vue [hunk 1]
```diff
@@ -340,6 +356,21 @@ defineExpose({
...
+  if (search === null) search = '';
+  set(counterparty, search);
+});
+
+const filteredCounterpartySuggestions: ComputedRef<string[]> = computed(() => {
+  const suggestions = get(counterparties);
+  const searchVal = get(counterpartySearch);
+  if (!searchVal) return suggestions;
+  return suggestions.filter(suggestion => suggestion.includes(searchVal));
+});
```
Comment (kelsos) [inline frontend/app/src/components/history/TransactionEventForm.vue:373]: Do you think we can extract a CounterPartySearch or something?

**Comment by kelsos** [comment]
@lukicenturi we also need to figure out why the tests are failing

File: frontend/app/src/components/history/TransactionEventForm.vue [hunk 2]
```diff
@@ -31,6 +31,8 @@ import { TaskType } from '@/types/task-type';
 import { One, Zero, bigNumberifyFromRef } from '@/utils/bignumbers';
 import { convertFromTimestamp, convertToTimestamp } from '@/utils/date';
 import { useEventTypeData } from '@/utils/history';
+import { useTransactions } from '@/store/history/transactions';
```
[the remark this answers is not in the document]
Comment (lukicenturi) [reply frontend/app/src/components/history/TransactionEventForm.vue:34]: Yes, it is

**Comment by kelsos** [comment]
@lukicenturi we also need a changelog entry here.

**Review by kelsos — APPROVED** [review]
lgtm @lukicenturi

**Pull request merged**